diff --git a/TestVectors/runtimes/net/Generated/DDBEncryption/TypeConversion.cs b/TestVectors/runtimes/net/Generated/DDBEncryption/TypeConversion.cs
index 1ff2d1b4..5e46a9aa 100644
--- a/TestVectors/runtimes/net/Generated/DDBEncryption/TypeConversion.cs
+++ b/TestVectors/runtimes/net/Generated/DDBEncryption/TypeConversion.cs
@@ -2236,11 +2236,11 @@ namespace AWS.Cryptography.DbEncryptionSDK.DynamoDb.Wrapped
             }
             if (value.NULL == true)
             {
-                return software.amazon.cryptography.services.dynamodb.internaldafny.types.AttributeValue.create_NULL(ToDafny_N3_com__N9_amazonaws__N8_dynamodb__S14_AttributeValue__M4_NULL(value.NULL));
+                return software.amazon.cryptography.services.dynamodb.internaldafny.types.AttributeValue.create_NULL(ToDafny_N3_com__N9_amazonaws__N8_dynamodb__S14_AttributeValue__M4_NULL((bool)value.NULL));
             }
-            if (value.IsBOOLSet)
+            if (value.BOOL != null)
             {
-                return software.amazon.cryptography.services.dynamodb.internaldafny.types.AttributeValue.create_BOOL(ToDafny_N3_com__N9_amazonaws__N8_dynamodb__S14_AttributeValue__M4_BOOL(value.BOOL));
+                return software.amazon.cryptography.services.dynamodb.internaldafny.types.AttributeValue.create_BOOL(ToDafny_N3_com__N9_amazonaws__N8_dynamodb__S14_AttributeValue__M4_BOOL((bool)value.BOOL));
             }
             throw new System.ArgumentException("Invalid Amazon.DynamoDBv2.Model.AttributeValue state");
         }
